pykafka/balancedconsumer.py

```diff
@@ -259,16 +260,19 @@ def stop(self):
         if self._consumer is not None:
             self._consumer.stop()
         self._zookeeper.remove_listener(self._zk_state_listener)
-        if self._owns_zookeeper:
-            # NB this should always come last, so we do not hand over control
-            # of our partitions until consumption has really been halted
-            self._zookeeper.stop()
-        else:
-            self._remove_partitions(self._get_held_partitions())
-            self._zookeeper.delete(self._path_self)
-            # additionally we'd want to remove watches here, but there are no
-            # facilities for that in ChildrenWatch - as a workaround we check
-            # self._running in the watcher callbacks (see further down)
+        try:
+            if self._owns_zookeeper:
+                # NB this should always come last, so we do not hand over control
+                # of our partitions until consumption has really been halted
+                self._zookeeper.stop()
+            else:
+                self._remove_partitions(self._get_held_partitions())
+                self._zookeeper.delete(self._path_self)
+                # additionally we'd want to remove watches here, but there are no
+                # facilities for that in ChildrenWatch - as a workaround we check
+                # self._running in the watcher callbacks (see further down)
+        except NoNodeError:
+            log.warning("Error encountered while closing ZooKeeper connection")
```

Comment:
(Just to check, I created a `BalancedConsumer` with an external `KazooClient` instance, and then called `stop()` on it twice - indeed the other lines here don't raise exceptions, just that one.)